subsys/net/l2/canbus/canbus.c

```diff
@@ -1002,6 +1082,191 @@ static enum net_verdict process_frame(struct net_pkt *pkt)
 	return ret;
 }
 
+#ifdef CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR
+static const u16_t mcast_addr = sys_cpu_to_be16(NET_CAN_MULTICAST_ADDR);
+
+static void eth_to_can_lladdr(struct net_pkt *pkt, bool bcast)
+{
+	struct net_eth_hdr *hdr = NET_ETH_HDR(pkt);
+	struct net_linkaddr *lladdr;
+
+	lladdr = net_pkt_lladdr_src(pkt);
+	lladdr->addr = hdr->src.addr;
+	lladdr->len = sizeof(struct net_eth_addr);
+	lladdr->type = NET_LINK_ETHERNET;
+
+	lladdr = net_pkt_lladdr_dst(pkt);
+	lladdr->len = sizeof(struct net_can_lladdr);
+	lladdr->type = NET_LINK_CANBUS;
+	if (bcast) {
+		lladdr->addr = (u8_t *)&mcast_addr;
+	} else {
+		lladdr->addr = &hdr->dst.addr[4];
+	}
+}
+
+static void forward_eth_frame(struct net_pkt *pkt)
+{
+	struct net_if *canbus_iface;
+
+	canbus_iface = net_if_get_first_by_type(&NET_L2_GET_NAME(CANBUS));
+	pkt->iface = canbus_iface;
+	net_if_queue_tx(canbus_iface, pkt);
+}
+
+static struct net_ipv6_hdr *get_ip_hdr_from_eth_frame(struct net_pkt *pkt)
+{
+	return (struct net_ipv6_hdr *)((u8_t *)net_pkt_data(pkt) +
+		sizeof(struct net_eth_hdr));
+}
+
+enum net_verdict net_canbus_translate_eth_frame(struct net_if *iface,
+						struct net_pkt *pkt)
+{
+	struct net_linkaddr *lladdr = net_pkt_lladdr_dst(pkt);
+	struct net_pkt *clone_pkt;
+
+	/*Forward only IPv6 frames*/
+	if ((get_ip_hdr_from_eth_frame(pkt)->vtc & 0xf0) != 0x60) {
+		return NET_CONTINUE;
+	}
+
+	/*This frame is for the Ethernet interface itself*/
+	if (net_linkaddr_cmp(net_if_get_link_addr(iface), lladdr) ||
+	    net_eth_is_addr_lldp_multicast((struct net_eth_addr *)lladdr->addr)) {
```

Comment:
Note sure I follow you here, what is the idea for checking lldp multicast address?

Reply:
LLDP is not supported by 6LoCAN